demo/src/com/google/maps/android/utils/demo/GeoJsonDemoActivity.java

```diff
@@ -0,0 +1,129 @@
+package com.google.maps.android.utils.demo;
+
+import com.google.android.gms.maps.model.BitmapDescriptor;
+import com.google.android.gms.maps.model.BitmapDescriptorFactory;
+import com.google.maps.android.geojson.GeoJsonFeature;
+import com.google.maps.android.geojson.GeoJsonLayer;
+import com.google.maps.android.geojson.GeoJsonPointStyle;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import android.os.AsyncTask;
+import android.util.Log;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+
+public class GeoJsonDemoActivity extends BaseDemoActivity {
+
+    private final static String mLogTag = "GeoJsonDemo";
+
+    private GeoJsonLayer mLayer;
+
+    // GeoJSON file to download
+    private final String mGeoJsonUrl
+            = "http://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/all_day.geojson";
+
+    protected int getLayoutId() {
+        return R.layout.geojson_demo;
+    }
+
+    @Override
+    protected void startDemo() {
+        DownloadGeoJsonFile downloadGeoJsonFile = new DownloadGeoJsonFile();
+        // Download the GeoJSON file
+        downloadGeoJsonFile.execute(mGeoJsonUrl);
+    }
+
+    /**
+     * Adds a point style to all features to change the color of the marker based on its magnitude property
+     */
+    private void addColorsToMarkers() {
+        // Iterate over all the features stored in the layer
+        for (GeoJsonFeature feature : mLayer.getFeatures()) {
+            // Check if the magnitude property exists
+            if (feature.hasProperty("mag")) {
+                double magnitude = Double.parseDouble(feature.getProperty("mag"));
+
+                // Get the icon for the feature
+                BitmapDescriptor pointIcon = BitmapDescriptorFactory.defaultMarker(magnitudeToColor(magnitude));
```

Comment:
Neat demo! I think you could take it a step further and use the IconGenerator to create different sized circles if you wanted to - totally not necessary though :)